### PR TITLE
App Service - Deploy server folder only

### PR DESCRIPTION
**Pull request by jasmingoh**
This change is required for deploying the server folder only.

Changes included:
1. Moved the .deployment file from the root to the server folder
2. Set the appService.deploySubpath to the server folder

**To test:**
1. Launch the extension in preview mode
2. Add App Service
3. Once App Service is deployed:
    - check the location of the .deployment file
    - check that `.vscode/settings.json` contains `appService.deploySubpath: "server"`
5. Do `npm install` and `npm build` (this will be automated in another PR)
4. Do a Ctrl + Shift + P, type in "Azure App Service: Deploy to Web App..."
5. Once deployment is done, click on "Browse website" and make sure that it is running properly

### CODE REVIEW

File: src/extension/src/constants.ts [hunk 1]
```diff
@@ -278,11 +278,13 @@ export const CONSTANTS = {
...
   APP_SERVICE_DEPLOYMENT: {
+    SERVER_FOLDER: "server",
     DOT_VSCODE_FOLDER: ".vscode",
     SETTINGS_FILE_NAME: "settings.json",
-    SETTINGS_FILE: (id: string) => {
+    SETTINGS_FILE: (id: string, subpath: string) => {
       return `{
-    "appService.defaultWebAppToDeploy": "${id}"
+    "appService.defaultWebAppToDeploy": "${id}",
+    "appService.deploySubpath":"${subpath}"
```
Comment (jcoc611-microsoft) [inline src/extension/src/constants.ts]: missing space after `:`? not running prettier?

Comment (jasmingoh) [reply src/extension/src/constants.ts]: oops, thanks, I do have format on save, but this was not caught because it is a string

**Review by jcoc611-microsoft — APPROVED** [review]

**Review by SahilTara — APPROVED** [review]

File: src/extension/src/constants.ts [hunk 2]
```diff
@@ -278,11 +278,13 @@ export const CONSTANTS = {
...
+    SERVER_FOLDER: "server",
     DOT_VSCODE_FOLDER: ".vscode",
     SETTINGS_FILE_NAME: "settings.json",
-    SETTINGS_FILE: (id: string) => {
+    SETTINGS_FILE: (id: string, subpath: string) => {
       return `{
-    "appService.defaultWebAppToDeploy": "${id}"
+    "appService.defaultWebAppToDeploy": "${id}",
+    "appService.deploySubpath": "${subpath}"
 }`;
```
Comment (jsondoo) [inline src/extension/src/constants.ts:288]: NIT: indentation

**Review by jsondoo — APPROVED** [review]
confirmed it deploy successfully 👍

**Pull request merged**